From 302dc7ba20478c545069080316e8e86c40ca8e0d Mon Sep 17 00:00:00 2001
From: Harini Katakam <harini.katakam@xilinx.com>
Date: Fri, 11 Jul 2014 14:15:00 +0530
Subject: [PATCH 413/793] spi: zynq-qspi: Wait for FIFO to be empty when
 switching TXD registers

When switching from TXD0 to TXD1/2/3, software has to wait for FIFO to be
empty before writing to TXD1/2/3 as per the documentation.

Although previous discussions and tests revealed that this waiting
is required only for TXD1/2/3 to TXD0 switches and not vice-verse,
design team conveyed that TXD0 to TXD1/2/3 switching can behave unpredictably
if wait is not used. Hence this change is made to comply.

Signed-off-by: Harini Katakam <harinik@xilinx.com>
Acked-by: Punnaiah Choudary Kalluri <punnaia@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/spi/spi-zynq-qspi.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/spi/spi-zynq-qspi.c b/drivers/spi/spi-zynq-qspi.c
index 6b38c38..d3ff687 100644
--- a/drivers/spi/spi-zynq-qspi.c
+++ b/drivers/spi/spi-zynq-qspi.c
@@ -544,7 +544,7 @@ static irqreturn_t zynq_qspi_irq(int irq, void *dev_id)
 				/* There is more data to send */
 				zynq_qspi_fill_tx_fifo(xqspi,
 						       ZYNQ_QSPI_RX_THRESHOLD);
-			} else {
+			} else if (intr_status & ZYNQ_QSPI_IXR_TXNFULL_MASK) {
 				int tmp;
 				tmp = xqspi->bytes_to_transfer;
 				zynq_qspi_copy_write_data(xqspi, &data,
-- 
1.7.5.4

